### ISSUE TITLE
[TSX - Parsing Error: '>' expected] in conjunction with @typescript-eslint/parser

### ISSUE THREAD

**Issue by stuartZhang** [vuejs/vue-eslint-parser#45]
Only if the TSX fragments are inside the *.vue file, the TSX parsing error always happens.

Provided the TSX fragments are absorbed into a single *.tsx file which is referenced in the vue file, everything is good.

My .eslintrc.js file is as such:
```javascript
module.exports = {
    parserOptions: {
        parser: '@typescript-eslint/parser',
        project: './tsconfig.json',
        tsconfigRootDir: './',
        ecmaFeatures: {
          jsx: true
        }
    },
    parser: 'vue-eslint-parser',
};
```
My tsconfig.jons is as below:
```json
 {    "compilerOptions": {
        "allowJs": true,
        "allowSyntheticDefaultImports": true,
        "baseUrl": "./",
        "checkJs": false,
        "esModuleInterop": true,
        "experimentalDecorators": true,
        "importHelpers": true,
        "isolatedModules": true,
        "jsx": "preserve",
        "jsxFactory": "h",
        "lib": [
            "es5",
            "es6",
            "es2015",
            "es7",
            "es2016",
            "es2017",
            "es2018",
            "esnext",
            "dom",
            "dom.iterable",
            "scripthost",
            "es2015.core",
            "es2015.collection",
            "es2015.generator",
            "es2015.iterable",
            "es2015.promise",
            "es2015.proxy",
            "es2015.reflect",
            "es2015.symbol",
            "es2015.symbol.wellknown",
            "es2016.array.include",
            "es2017.object",
            "es2017.intl",
            "es2017.sharedmemory",
            "es2017.string",
            "es2017.typedarrays",
            "es2018.intl",
            "es2018.promise",
            "es2018.regexp",
            "esnext.asynciterable",
            "esnext.array",
            "esnext.intl",
            "esnext.symbol"
        ],
        "module": "esnext",
        "moduleResolution": "node",
        "noEmit": true,
        "noImplicitAny": false,
        "paths": {
            "@/*": ["src/*"]
        },
        "skipLibCheck": false,
        "sourceMap": true,
        "strict": true,
        "strictNullChecks": true,
        "strictPropertyInitialization": true,
        "target": "esnext"
    },
    "include": [
        "node_modules/mx-general.macros/types/*.d.ts",
        "node_modules/log2indexeddb/types/*.d.ts",
        "types/*.d.ts",
        "src/**/*.vue",
        "src/**/*.ts",
        "src/**/*.tsx",
        "src/**/*.js",
        "src/**/*.jsx"
    ]
}
```
My vue file is as follows:
```html
<template>
    <div>
        <tree ref="domainTree" class="tree" :data="domains" :render="renderTreeNode" />
    </div>
</template>
<script lang="tsx">
```
```typescript
import pcLog from 'mx-general.macros/dist/log-sync.macro';
import {mapState} from 'vuex';
import icon from '../assets/images/favicon.ico';
export default {
    name: 'IViewDemo',
    data(){
        return {};
    },
    computed: {
        ...mapState(['domains']),
        renderTreeNode(){
            const domainId = 0;
            const render: iView2.TreeRender = (h, {data}) => {
                let clazz = 'ivu-tree-title';
                if (data.id === domainId) {
                    clazz += ' ivu-tree-title-selected';
                }
                const onClick = () => {
                    if (data.disabled) {
                        return;
                    }
                    (this.$refs.domainTree as iView2.Tree).handleSelect(data.nodeKey);
                    if (data.id !== domainId) {
                        pcLog.warn(`选中：${data.name}`);
                        alertAsync(`选中：${data.name}`);
                    }
                };
                if (data.leaf) {
                    return <span class={clazz} onClick={onClick}><img src={icon} /> {data.name}</span>; // Parsing error: '>' expected
                }
                return <span class={clazz} onClick={onClick}>{data.name}</span>;
            }
            return render;
        }
    },
    mounted(){
        this.$store.dispatch('getDomainIds');
    }
};
```
```html
</script>
<style scoped>
.tree >>> .ivu-tree-title > img {
    width: 1rem;
    height: 1rem;
    vertical-align: middle;
}
</style>
```

![未命名](https://user-images.githubusercontent.com/13935927/58445553-aa2c8080-812f-11e9-88a0-b2da7c6d9b80.png)

**Comment by mysticatea** [comment]
I could not reproduce it. What is the version of the parsers you are using?

**Comment by stuartZhang** [comment]
1. I have uploaded my demo project to there: http://dev8.dehuinet.com:8018/web_apps/mobile_attendance/static/ts-admin-iview-demo2.zip
2. Please you download and open it in VSCode. Since All dependencies are in place, it's unnecessary to "npm i".
3. The core ESLint configuration file is inside "node_modules/eslint-config-minxing/eslint-config-vue-ts.js"
4. The ESLint configuration file "src/.eslintrc.js" merely references "node_modules/eslint-config-minxing/eslint-config-vue-ts.js"
5. The observed *.vue file is "src/components/IViewDemo.vue"
6. Go to the project root directory and execute "npx eslint src/components/IViewDemo.vue". The reported bug can be reproduced.

Otherwise, my machine is "Windows 10". If you'd like to re-install dependencies in person, the below steps are prerequisite:

1. npm i -g nrm
2. nrm add dehuinet http://npm.dehuinet.com:8100
3. nrm use dehuinet
4. npm i

**Comment by stuartZhang** [comment]
If the TS codes are migrated to a single "IViewDemo.ts" file which is referenced in "IViewDemo,vue" by means of "<script lang="ts" src="./IViewDemo.ts">", the parsing error for the 35th line will disappear.

**Comment by mysticatea** [comment]
Thank you. But I could not access to your zip.

![image](https://user-images.githubusercontent.com/1937871/58464612-fc41c600-8170-11e9-8f47-3f921f3114e6.png)

I'm asking "What is the version of the parsers you are using?" because some versions of `@typescript-eslint/parser` doesn't support `ecmaFeatures.jsx` option. You can check the versions with the following command:

```
npm ls @typescript-eslint/parser
```

**Comment by stuartZhang** [comment]
The version of the parsers I'm using is as follows:

$ npm ls @typescript-eslint/parser
ts-admin-iview-demo2@1.0.0 D:\workshop\github\temp1\ts-admin-iview-demo2
+-- @typescript-eslint/parser@1.9.0
`-- eslint-config-minxing@1.1.9
  `-- @typescript-eslint/parser@1.9.0  deduped

**Comment by mysticatea** [comment]
Thank you.

Hmm. Sounds fine, it's the latest version.

What does the following command print?

```
npx eslint --print-config src/components/IViewDemo.vue
```

**Comment by stuartZhang** [comment]
The output of attaching --print-config option is in the log file.

[log.txt](https://github.com/mysticatea/vue-eslint-parser/files/3226835/log.txt)

**Comment by stuartZhang** [comment]
Moreover, I'm uploading the zip file to the Google Drive. And then, I plan to share it without any Auth.

**Comment by mysticatea** [comment]
Thank you. That log.txt looks fine. Hmm...

**Comment by stuartZhang** [comment]
Please you download the zip file from https://drive.google.com/open?id=1QoOD5f2kS15n5dIbMf6i2ARgTylstjQl

**Comment by mysticatea** [comment]
Thank you. I could reproduce it with your zip. I'm investigating...

**Comment by mysticatea** [comment]
Thank you. I have found that `@typescript-estree/typescript-estree` package ignores `jsx` option if `project` option existed. See https://github.com/typescript-eslint/typescript-eslint/issues/566.

**Comment by stuartZhang** [comment]
You're great. I shall keep track of the new issue.

**Comment by Merott** [comment]
Is there a solution/workaround to this? I'm having the same issue in a Vue file that uses `<script lang="tsx">`

**Comment by wenkanglin** [comment]
+1, any update?

**Comment by wesam-k** [comment]
1+ any solution

**Comment by SunShinewyf** [comment]
any update?

**Comment by ota-meshi** [comment]
The issue opened in typescript-eslint seems to be resolved, so I close this issue.
If you have any problems, please open a new issue or follow another related issue.

**Comment by ldwonday** [comment]
@stuartZhang 你这个问题最后怎么修复的？

**Comment by wztlink1013** [comment]
所以说了这么多，这个问题是怎么修复的？

**Comment by CHENQIKUAI** [comment]
![image](https://user-images.githubusercontent.com/25342395/145180213-679b636e-1705-48cb-9a83-3d9e446ee289.png)

**Comment by Newfffff** [comment]
tks a lot

**Comment by EliazTray** [comment]
https://eslint.vuejs.org/user-guide/#using-jsx

**Comment by carter-1uo** [comment]
version:
"eslint": "^9.22.0",
"eslint-plugin-vue": "^10.0.0",
"typescript-eslint": "^8.26.1",

When i set
```js
{
    //...
    parserOptions: {
         project: './tsconfig',
         ecmaFeatures: { jsx: true },
    }
}
```
this will cause same error as 'jsx : false'.

After removing 'project', it parses jsx&tsx well. But cause another error: Some global types like 'ChildNode' marked as undef by eslint.
Disable 'no-undef' for "*.vue"  help(ugly though).

Hope these information help.

**Comment by eric-gitta-moore** [comment]
same here.
```
  eslintrc:ignore-pattern Check {
  filePath: '/Users/aaa/workspace/xxx/robot/Robot.vue',
  dot: false,
  relativePath: 'src/xxx/robot/Robot.vue',
  result: false
} +1ms
  eslint:cli-engine Lint /Users/aaa/workspace/xxx/robot/Robot.vue +0ms
  eslint:linter Linting code for /Users/aaa/workspace/xxx/robot/Robot.vue (pass 1) +0ms
  eslint:linter Verify +0ms
  eslint:linter With ConfigArray: /Users/aaa/workspace/xxx/robot/Robot.vue +0ms
  eslint:linter Apply the processor: 'vue/.vue' +0ms
  eslint:linter A code block was found: '(unnamed)' +0ms
  eslint:linter Parsing: /Users/aaa/workspace/xxx/robot/Robot.vue +1ms
  eslint:linter Parsing error: '>' expected.
  eslint:linter SyntaxError: '>' expected.
    at ParseError.normalize (/Users/aaa/workspace/.eden-mono/temp/node_modules/.pnpm/vue-eslint-parser@10.2.0_eslint@8.44.0/node_modules/vue-eslint-parser/index.js:76:20)
    at parseScript (/Users/aaa/workspace/.eden-mono/temp/node_modules/.pnpm/vue-eslint-parser@10.2.0_eslint@8.44.0/node_modules/vue-eslint-parser/index.js:5384:33)
    at parseScriptSetupElements (/Users/aaa/workspace/.eden-mono/temp/node_modules/.pnpm/vue-eslint-parser@10.2.0_eslint@8.44.0/node_modules/vue-eslint-parser/index.js:5416:20)
    at parseAsSFC (/Users/aaa/workspace/.eden-mono/temp/node_modules/.pnpm/vue-eslint-parser@10.2.0_eslint@8.44.0/node_modules/vue-eslint-parser/index.js:6580:18)
    at Object.parseForESLint (/Users/aaa/workspace/.eden-mono/temp/node_modules/.pnpm/vue-eslint-parser@10.2.0_eslint@8.44.0/node_modules/vue-eslint-parser/index.js:6534:53)
    at parse (/Users/aaa/workspace/.eden-mono/temp/node_modules/.pnpm/eslint@8.44.0/node_modules/eslint/lib/linter/linter.js:809:22)
    at Linter._verifyWithoutProcessors (/Users/aaa/workspace/.eden-mono/temp/node_modules/.pnpm/eslint@8.44.0/node_modules/eslint/lib/linter/linter.js:1285:33)
    at /Users/aaa/workspace/.eden-mono/temp/node_modules/.pnpm/eslint@8.44.0/node_modules/eslint/lib/linter/linter.js:1812:29
    at Array.map (<anonymous>)
    at Linter._verifyWithProcessor (/Users/aaa/workspace/.eden-mono/temp/node_modules/.pnpm/eslint@8.44.0/node_modules/eslint/lib/linter/linter.js:1807:37) +8s
  eslint:linter Generating fixed text for /Users/aaa/workspace/xxx/robot/Robot.vue (pass 1) +0ms
  eslint:source-code-fixer Applying fixes +0ms
  eslint:source-code-fixer shouldFix parameter was false, not attempting fixes +0ms
  eslint:file-enumerator Complete iterating files: ["/Users/aaa/workspace/xxx/robot/Robot.vue"] +9s
  eslint:cli-engine Linting complete in: 9156ms +8s

/Users/aaa/workspace/xxx/robot/Robot.vue
  189:12  error  Parsing error: '>' expected

✖ 1 problem (1 error, 0 warnings)
```
